driven/storage/adapter_custom.go

```diff
@@ -0,0 +1,951 @@
+		return errors.WrapErrorAction(logutils.ActionInsert, "error inserting contents", nil, err)
+	}
+	return nil
+}
+
+// customContentConversionAPIToStorage formats API struct to stroage struct
+func (sa *Adapter) customContentConversionAPIToStorage(item model.Content) content {
+	//parse into the storage format and pass parameters
+	var extractedKey []string
+	for _, val := range item.LinkedContent {
+		extractedKey = append(extractedKey, val.Key)
+	}
+
+	var content content
+	content.ID = item.ID
+	content.AppID = item.AppID
+	content.OrgID = item.OrgID
+	content.Key = item.Key
+	content.Type = item.Type
+	content.Name = item.Name
+	content.Details = item.Details
+	content.ContentReference = item.ContentReference
+	content.LinkedContent = extractedKey
+	content.DateCreated = item.DateCreated
+	content.DateUpdated = item.DateUpdated
+
+	return content
+}
+
+// UpdateCustomContent updates a content
+func (sa *Adapter) UpdateCustomContent(key string, item model.Content) error {
+	//parse into the storage format and pass parameters
+	var extractedKey []string
+	for _, val := range item.LinkedContent {
+		extractedKey = append(extractedKey, val.Key)
+	}
+
+	filter := bson.M{"org_id": item.OrgID, "app_id": item.AppID, "key": key}
+	update := bson.M{
+		"$set": bson.M{
+			"type":           item.Type,
+			"details":        item.Details,
+			"name":           item.Name,
+			"reference":      item.ContentReference,
+			"linked_content": extractedKey,
+			"date_updated":   time.Now(),
+		},
+	}
+	result, err := sa.db.customContent.UpdateOne(sa.context, filter, update, nil)
+	if err != nil {
+		return errors.WrapErrorAction(logutils.ActionUpdate, "", &logutils.FieldArgs{"key": key}, err)
+	}
+	if result.MatchedCount == 0 {
+		return errors.WrapErrorData(logutils.StatusMissing, "", &logutils.FieldArgs{"key": key}, err)
+	}
+	return nil
+}
+
+// DeleteCustomContent deletes a content
+func (sa *Adapter) DeleteCustomContent(appID string, orgID string, key string) error {
+	filter := bson.M{"org_id": orgID, "app_id": appID, "key": key}
+	result, err := sa.db.customContent.DeleteOne(sa.context, filter, nil)
+	if err != nil {
+		return errors.WrapErrorAction(logutils.ActionDelete, "", &logutils.FieldArgs{"key": key}, err)
+	}
+	if result == nil {
+		return errors.WrapErrorData(logutils.StatusInvalid, "result", &logutils.FieldArgs{"key": key}, err)
+	}
+	deletedCount := result.DeletedCount
+	if deletedCount == 0 {
+		return errors.WrapErrorData(logutils.StatusMissing, "", &logutils.FieldArgs{"key": key}, err)
+	}
+	return nil
+}
+
+// DeleteContentKeyFromLinkedContents deletes a content key from linkedContent field within customContent collection
+func (sa *Adapter) DeleteContentKeyFromLinkedContents(appID string, orgID string, key string) error {
+	var keyArr []string
+	keyArr = append(keyArr, key)
+	filter := bson.M{"org_id": orgID, "app_id": appID}
+	//delete key from linked_content
+	update := bson.M{
+		"$pull": bson.M{
+			"linked_content": bson.M{
```

Comment:
Only one key is being removed, so the key can be set as the value for `"linked_content"` directly. The same thing should apply to all of the functions below where you are using the `$pull` operation to remove a key.

Reply:
for example, 
"$pull": bson.M{
     "linked_content": key,
} ? 

I have this chunk of code comment out under these lines and it might because it doesn't work. Will test it out to see if it works